R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -2070,6 +2072,16 @@ test_that("dapply() on a DataFrame", {
   rownames(expected) <- NULL
   expect_identical(expected, result)
 
+  result <- dapplyCollect(
+              df,
+              function(x) {
+                y <- x[x$a > 1, ]
+                y <- cbind(y, y$a + 1L)
```

Comment:
Maybe, we can add test cases where the function returns data.frame, list and other data structures. What do you think ? 
In general, it seems that we only support a subset of R's data types and there might be issues when we train a model on a worker and try to pass that model to a predict to make predictions using a separate dapply method.